controllers/imagepolicy_controller.go

```diff
@@ -127,7 +127,15 @@ func (r *ImagePolicyReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error)
 	if policer != nil {
 		tags, err := r.Database.Tags(repo.Status.CanonicalImageName)
 		if err == nil {
-			latest, err = policer.Latest(tags)
+			var filter *policy.RegexFilter
+			if pol.Spec.FilterTags != nil {
+				filter, err = policy.NewRegexFilter(pol.Spec.FilterTags.Pattern, pol.Spec.FilterTags.Extract)
+				if err != nil {
+					return ctrl.Result{}, err
+				}
+				filter.Apply(tags)
+			}
+			latest, err = policer.Latest(tags, filter)
```

Comment:
Running the flow of control the other way would be clearer here. This way around, you supply a filter to the `Policer`, and it has to know how to deal with it. But if you filter here, and give the `Policer` the extracted names, it can determine which one it wants without having to know whether they are filtered or not. Then you can map back to the original again here (if there was a filter in the first place).

Reply:
One issue, whichever way around you do it, is what to do if the extracted strings are duplicated? E.g., if you have `(dev|stage)-(.*)`, and extract `$2`, you will get a duplicate from the tags `{"dev-v1.0", "stage-v1.0"}`. At present the last one encountered wins (since RegexpFilter keeps track of them in a map). But this isn't stable -- you can get different results, without anything actually changing.

Reply:
Thanks for the feedback, I think your suggestion makes a lot of sense and would simplify things significantly in the policies `Latest` implementation.

As for the second comment, I thought about this as well when writing the tests, I wasn't able to think of a real-world scenario where this would become an issue but I think it does make sense to specify this in the documentation since it's indeed an edge case some might stumble upon. I don't know if we can do anything about this conflict aside from warning users about it.

Reply:
> it does make sense to specify this in the documentation since it's indeed an edge case some might stumble upon

I agree with this, and

> warning users about it

with this. If it happens, it could be pretty mysterious, so we could log a message with a reference to docs or decent search term.

Reply:
(we can discuss these alternatives a bit more in https://github.com/fluxcd/image-reflector-controller/issues/77)